src/planwise/client/projects2/components/settings.cljs

```diff
@@ -125,92 +145,134 @@
-          [m/TextFieldHelperText {:persistent true} (str "Percentage of population that should be considered " (get-in @current-project [:config :demographics :unit-name]))]]
-
-         [:section {:class-name "project-settings-section"}
-          [section-header 3 "Providers"]
-          [providers-set-dropdown-component {:label     "Provider Set"
-                                             :value     (:provider-set-id @current-project)
-                                             :on-change #(dispatch [:projects2/save-key :provider-set-id %])
-                                             :disabled? read-only}]
-
-          [current-project-input "Capacity workload" [:config :providers :capacity] "number" {:disabled read-only :sub-type :float}]
-          [m/TextFieldHelperText {:persistent true} (str "How many " (get-in @current-project [:config :demographics :unit-name]) " can be handled per provider capacity")]
-
-          (when-not read-only [tag-input])
-          [:label "Tags: " [tag-set @tags read-only]]
-          [count-providers @tags @current-project]]
-
-         [:section {:class-name "project-settings-section"}
-          [section-header 4 "Coverage"]
-          [coverage-algorithm-filter-options {:coverage-algorithm (:coverage-algorithm @current-project)
-                                              :value              (get-in @current-project [:config :coverage :filter-options])
-                                              :on-change          #(dispatch [:projects2/save-key [:config :coverage :filter-options] %])
-                                              :empty              [:div {:class-name " no-provider-set-selected"} "First choose provider-set."]
-                                              :disabled?          read-only}]]
-
-         [:section {:class-name "project-settings-section"}
-          [:div [:p [m/Icon "account_balance"] "Available budget"]]
-          [current-project-input "" [:config :actions :budget] "number" {:disabled read-only :class "project-setting"}]
-          [m/TextFieldHelperText {:persistent true} "Planwise will keep explored scenarios below this maximum budget"]
-
-          [:div [:p [m/Icon "domain"] "Building a new provider..."]]
-          [listing-actions {:read-only?  read-only
-                            :action-name :build
-                            :list        @build-actions}]
-
-          [:div [:p [m/Icon "arrow_upward"] "Upgrading a provider so that it can satisfy demand would cost..."]]
-          [current-project-input "" [:config :actions :upgrade-budget] "number" {:disabled read-only :class "project-setting"}]
-
-          [:div [:p [m/Icon "add"] "Increase the capactiy of a provider by..."]]
-          [listing-actions {:read-only?   read-only
-                            :action-name :upgrade
-                            :list        @upgrade-actions}]]]]])))
+         (case step
```

Comment:
You can put the section component in the data structure describing the sections above as well and you won't need this case.